backend-app/controllers/users_controllers/admin_controller.ts

```diff
@@ -197,92 +137,117 @@ export const unbanUser = async (req: IReq, res: IRes, next: INext) => {
             throw new AppError(400, 'User is not banned');
         user.accessRestricted = false;
         await user.save();
-        res.status(200).json({
+        this.setStatus(200);
+        return {
             message: 'User is now unbanned',
-        });
-    } catch (err) {
-        next(err);
+        };
     }
-};
 
-export const createRole = async (req: IReq, res: IRes, next: INext) => {
-    const { name, authorities, restrictions } = req.body;
-    try {
+    @Response(400, 'Role already exists')
+    @SuccessResponse('201', 'CREATED')
+    @Middlewares(restrictTo(Actions.MANAGE_ROLES))
+    @Security('jwt')
+    @Post('role')
+    async createRole(
+        @Body()
+        body: RoleType
+    ): Promise<{ message: string; data: RoleType }> {
+        const { name, authorities, restrictions } = body;
         if (await Role.getRoleByName(name))
             throw new AppError(400, 'Role already exists');
         const createdRole = await Role.createRole(
             name,
             authorities,
             restrictions
         );
-        res.status(201).json({
+        this.setStatus(201);
+        return {
             message: 'Role created',
             data: createdRole,
-        });
-    } catch (err) {
-        next(err);
+        };
     }
-};
-
-export const getRoles = async (_req: IReq, res: IRes, next: INext) => {
-    try {
+    @SuccessResponse('200', 'OK')
+    @Middlewares(restrictTo(Actions.MANAGE_ROLES))
```

Comment:
Remember this because you will need to change it in the next paragraph because I've created a decorator for this: @inspectAuthority("MANAGE_ROLES")

Reply:
okay